runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1300,51 +1359,83 @@ TR::Node *TR_VectorAPIExpansion::transformStoreToArray(TR_VectorAPIExpansion *op
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, numLanes, mode, 1, Other);
    }
 
 TR::Node *TR_VectorAPIExpansion::binaryIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
-                                                        TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+                                                        TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes,
+                                                        handlerMode mode)
    {
-   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, mode, 2, Other);
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, numLanes, mode, 2, Other);
    }
 
 TR::Node *TR_VectorAPIExpansion::reductionCoercedIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
-                                                                  TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+                                                                  TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes,
+                                                                  handlerMode mode)
    {
-   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, mode, 1, Reduction);
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, numLanes, mode, 1, Reduction);
    }
 
 TR::Node *TR_VectorAPIExpansion::ternaryIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
-                                                        TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+                                                         TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes,
+                                                         handlerMode mode)
    {
-   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, mode, 3, Other);
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, numLanes, mode, 3, Other);
    }
 
 TR::Node *TR_VectorAPIExpansion::naryIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
-                                                      TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode,
+                                                      TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes,
+                                                      handlerMode mode,
                                                       int32_t numChildren, vapiOpCodeType opCodeType)
    {
    TR::Compilation *comp = opt->comp();
    TR::Node *opcodeNode = node->getFirstChild();
+   int firstOperand = 5;
+   TR::Node *maskNode = node->getChild(firstOperand + numChildren);
```

Comment:
how are we sure there is a non-NULL maskNode?  i guessed this routine is handling non-masked unary vector operations too, right?

Reply:
if maskNode can be NULL, it seemed to me it will crash on next line

Reply:
Each intrinsic node has a child that represents the mask. For non-masked operations it's `aconst NULL`. So there wont be a crash here.

Reply:
I will add a comment about that.

Reply:
Done